api/filter/tcp_proxy.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+package envoy.api.v2.filter;
+
+import "google/protobuf/duration.proto";
+
+// [V2-API-DIFF] The route match now takes place in the FilterChainMatch table.
+message TcpProxy {
+  // The human readable prefix to use when emitting statistics for the
+  // TCP proxy filter. See the statistics documentation for more information.
+  string stat_prefix = 1;
+
+  // The upstream cluster to connect to.
+  string cluster = 2;
```

Comment:
Canary, BG, etc. I think it's worthwhile to have it here, but can definitely be done in follow up since it's an extension to what we have today. We should share the weighted cluster Message with RDS also.